site/src/pages/api/auth/signout.ts

```diff
@@ -0,0 +1,12 @@
+import type { APIRoute } from 'astro';
+import { createSupabaseServerClient } from '../../../lib/supabase-server';
+
+export const POST: APIRoute = async ({ cookies, redirect }) => {
+  const supabase = createSupabaseServerClient(cookies);
+
+  // Sign out from Supabase (this automatically clears cookies)
+  await supabase.auth.signOut();
```

Comment:
Attaching the server client to locals in the middleware would also make this neater

```suggestion
export const POST: APIRoute = async ({ cookies, redirect, locals }) => {
  await locals.auth.signOut();
```